sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -315,6 +315,57 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSparkSession {
     checkAnswer(df.select(isnotnull(col("a"))), Seq(Row(false)))
   }
 
+  test("Test NULLIF function") {
+
+    def testNullIfFunction(alwaysInlineCommonExpr: String): Unit = {
+      withSQLConf(SQLConf.ALWAYS_INLINE_COMMON_EXPR.key -> alwaysInlineCommonExpr) {
+        val spark = SparkSession.builder().getOrCreate()
+
+        val result1 = spark.sql("SELECT NULLIF(1, 1)").collect().head.get(0)
+        assert(result1 == null)
```

Comment:
Could you simplify it, and re-use `checkAnswer`:
```suggestion
    checkAnswer(sql("SELECT NULLIF(1, 1)"), Row(null))
```